platform-sdk/platform-apps/tests/ConsistencyTestingTool/src/main/java/com/swirlds/demo/consistency/ConsistencyTestingToolState.java

```diff
@@ -284,12 +303,24 @@ public void handleConsensusRound(
 
         roundsHandled++;
 
-        round.forEachTransaction(this::applyTransactionToState);
+        round.forEachEventTransaction((ev, tx) -> {
+            applyTransactionToState(ev, tx, stateSignatureTransaction);
```

Comment:
You can filter system transactions before entering applyTransactionToState and avoid passing a 3rd argument.

Something like this, inside the lambda body:


if (isSystemTransaction(tx)) {
                stateSignatureTransaction.accept(
                        new ScopedSystemTransaction(ev.getCreatorId(), ev.getSoftwareVersion(), tx));
                return;
   } else {
            applyTransactionToState(ev, tx);
  }